content/things/observables-on-the-platform.md

```diff
@@ -0,0 +1,375 @@
+
+### Microtask boundaries
+
+When talking to [Jake Archibald] about this, he pointed out that there might be a difference in task timing. And of course, he turned out to be right. What a jerk.
+
+Here’s two more pieces of code that _look_ the equivalent, but behave differently:
+
+```js
+import { Observable } from "rxjs";
+
+new Observable(subscriber => {
+  console.log("before");
+  subscriber.next("value");
+  console.log("after");
+}).subscribe(value => {
+  console.log("received:", value);
+});
+```
+
+Running this RxJS code, you will see the following logs:
+
+- `before`
+- `received: value`
+- `after`
+
+```js
+import { fromNext, subscribe } from "observables-with-streams";
+
+fromNext(next => {
+  console.log("before");
+  next("value");
+  console.log("after");
+}).pipeTo(
+  subscribe(value => {
+    console.log("received:", value);
+  })
+);
+```
+
+With ows, you will see these logs:
+
+- `before`
+- `after`
+- `received: value`
+
+To put it in a sentence: **RxJS observables deliver their values _synchronously_ to subscribers. Streams run their processing steps in a microtask, making them _asynchronous_.**
+
+## ows for app development
+
+Now that I had established where streams (and by extension ows) work differently than RxJS, I was wondering if these differences make ows impractical for writing web apps. Time to write a little example app. Luckily, I had an itch that I needed to scratch:
+
+When you take a picture with your camera, the camera needs to focus on a subject. The “focus point”, the point in space that the camera is focusing on, is often shown on the screen of your camera. But not only that specific point is in focus. Subjects closer to the camera (and subjects further away) can also appear sharp on the picture that you take, depending on _how much_ they are deviating from the focus point. The region that subjects can move around in and still remain in focus is called the “Depth of Field”, or DoF for short. Its size depends on a number of things: Focal length and aperture of the lens, subject distance and sensor size of the camera to begin with. There a number of apps out there that calculate the DoF for you based on these variables, but some have a disappointing UX or only expose a subset of the data I am interested in.
+
+As any self-respecting web developer who technically had other, more pressing responsibilities, I procrastinated by writing my own app. The tool that came out of this effort is called [DoF Tool]. Another uninspiring name. DoF Tool is [open source][dof source] and makes use of [observables-with-streams][ows] for all of the UI and user interactions. ows is highly tree-shakable, so it should be used with a bundler. For DoF Tool I’m using [Rollup]. However, if you just want to take ows for a quick spin to try it out, you can include it from a CDN like [JSDelivr] as one big bundle:
+
+```html
+<script src="https://cdn.jsdelivr.net/npm/observables-with-streams@latest/dist/really-big-bundle.js"></script>
```

Comment:
Why not modules and unpkg?

Reply:
JSDelivr is run by Cloudflare and has proven more stable.

I only provide one `really-big-bundle.js` and it’s UMD. I chose UMD here because I wanted it to also work in workers (and not just in Chrome).